public/locations-list-map.js

```diff
@@ -399,86 +465,147 @@ function initMap(states) {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      // Use navigator provided lat/long coords to center map now.
+      centerMapToMarkersNearCoords(position.coords.latitude, position.coords.longitude);
+
+    }, function (err) {
+      // Hide the "User my location" link since we know that will not work.
+      $('#use-location').hide();
+
+    }, {
+      maximumAge: Infinity,
+      timeout: 10000
+    });
+  }
+}
 
-                markerDistances[distance] = marker;
 
-                distances.push(distance);
-            }
 
-            // sort the distances and set bounds to closest three
-            distances.sort((a, b) => a - b);
+/**
+ * Centers map around markers nearest to an arbitrary set of latitude/longitude coordinates.
+ */
+function centerMapToMarkersNearCoords(latitude, longitude) {
+  var latlng = new google.maps.LatLng(latitude, longitude);
+
+  //Compute the distances of all markers from the user
+  var markerDistances = new Map(); // an associative array containing the marker referenced by the computed distance
+  var distances = []; // all the distances, so we can sort and then call markerDistances
+  for (const marker of markers) {
+    let distance = google.maps.geometry.spherical.computeDistanceBetween(marker.position, latlng);
 
+    // HACK: In the unlikely event that the exact same distance is computed, add one meter to the distance to give it a unique distance
+    // This could occur if a marker was added twice to the same location.
+    if (markerDistances.has(distance)) { distance = distance + 1; }
 
-            // center the map on the user
-            const bounds = new google.maps.LatLngBounds();
-            bounds.extend(user_latlng);
+    markerDistances[distance] = marker;
 
-            // Extend the bounds to contain the three closest markers
-            let i = 0;
-            while (i < 3) {
-                // Get one of the closest markers
-                var distance = distances[i]
-                var marker = markerDistances[distance];
+    distances.push(distance);
+  }
 
-                // Add to the iterator first just in case something fails later to avoid infinite loop
-                i++;
+  // sort the distances and set bounds to closest three
+  distances.sort((a, b) => a - b);
 
-                const marker_lat = marker.position.lat();
-                const marker_lng = marker.position.lng();
+  // center the map on the user
+  const bounds = new google.maps.LatLngBounds();
+  bounds.extend(latlng);
 
-                const loc = new google.maps.LatLng(marker_lat, marker_lng);
-                bounds.extend(loc);
-            }
-            map.fitBounds(bounds);       // auto-zoom
-            map.panToBounds(bounds);     // auto-center
-        })
-    }
+  // Extend the bounds to contain the three closest markers
+  let i = 0;
+  while (i < 3) {
```

Comment:
Why not use a for loop?

`for (let i = 0; i < 3; i++)`

?

Reply:
I didn’t write this, only a refactor. Unfortunately lots of this is showing up as me due to relocating or indenting. But yeah...

Reply:
fixed just because I'm ocd